### ISSUE TITLE
Parser gap: 'function name;' with no body is rejected

### ISSUE THREAD

**Issue by ss-o** [z-shell/zsh-lint#351]
## Summary

`function name;` with nothing after it is rejected, but native Zsh accepts it.

```zsh
function a;
```

```
t.zsh:1:1: `foo()` must be followed by a statement
```

`zsh -f -n` is clean, and running it is a no-op: no function is defined and no error is raised.

```
$ zsh -f -c 'function a;
functions a'
```

prints nothing — `a` does not exist afterwards.

## Not the non-brace body gap

Fixed in #349, which deliberately left this failing: every shape fixed there has a real body, and the body gate refuses end-of-input precisely so `a () ;` stays rejected. That rejection is correct for the `()` spelling — Zsh rejects `a () ;` — but the keyword spelling differs.

| Source | zsh | zsh-lint |
| --- | --- | --- |
| `a () ;` | rejected | rejected (correct) |
| `function a;` | accepted, defines nothing | rejected |

## Zsh manual section

zshmisc, *Shell Grammar* — the `function` reserved word's handling of an empty definition.

## Acceptance criteria

- `function a;` parses.
- `a () ;` stays rejected, matching Zsh.
- A fixture demonstrating that no function is defined.

## Version

zsh-lint at 6fca990 (main).
